src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,108 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+const pattern =
+  // eslint-disable-next-line max-len
+  /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [img, setImg] = useState('');
+  const [imdbLink, setImdbLink] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const isValid = () => {
+    const isFilled = [title, img, imdbLink, imdbId].every(
+      field => field.trim() !== '',
```

Comment:
```suggestion
      field => field.trim(),
```
You don't need to compare with an empty string, just check if a value exists